ENIGMAsystem/SHELL/Universal_System/estring.cpp

```diff
@@ -383,4 +393,27 @@ var string_split(const std::string &str, const std::string &delim,
   return res;
 }
 
+string directory_contents(string dname) {
+  if (string_replace_all(dname, " ", "") == "") dname = ".";
+  if (directory_exists(dname)) {
+    string rpath = filename_absolute(dname);
+    #if CURRENT_PLATFORM_ID == OS_WINDOWS
```

Comment:
This is why we need `file_path_join`. Doing operating system checks in every file in the engine isn't good.